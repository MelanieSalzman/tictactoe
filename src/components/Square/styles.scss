.square {
  background: #ffffff90;
  border: 0.1rem solid rgb(36, 5, 107);
  float: left;
  font-size: 24px;
  font-weight: bold;
  line-height: 34px;
  height: 5em;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 5em;
  border-radius: 5px;
}

.square:focus {
  outline: none;
}

.kbd-navigation .square:focus {
  background: #ddd;
}

.planetIcon {
  width: 3rem;
  height: 3rem;
}
