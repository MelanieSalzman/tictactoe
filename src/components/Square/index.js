import React from 'react'
import './styles.scss'
import saturn from '../../assets/images/saturn.png'
import earth from '../../assets/images/earth.png'

const Square =
  ({ player, onClick }) => {
    const getPlayerIcon = (player) => {
      return player === 'X' ? saturn : earth
    }

    return (
      <button
        className='square'
        onClick={onClick}
      >
        {player && <img src={getPlayerIcon(player)} alt={player} className='planetIcon'/>}
      </button>
    )
  }

export default Square
