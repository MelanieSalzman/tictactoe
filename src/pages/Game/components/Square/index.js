import React from 'react'
import './styles.scss'
import PlanetIcon from '../PlanetIcon'


const Square =
  ({ value, onClick }) => (
    <button
      className="square"
      onClick={onClick}
    >
      {<PlanetIcon player={value} />}
    </button>
  )

export default Square
