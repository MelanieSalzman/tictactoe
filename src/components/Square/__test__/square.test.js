import React from 'react'
import ReactDOM from 'react-dom'
import Square from '..'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('Square renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Square />, div)
})

it('Square renders correctly', () => {
  const { getByTestId } = render(<Square value='X' />)
  expect(getByTestId('square')).toHaveClass('square')
})

it('Square matches snapshot', () => {
  const tree = renderer.create(<Square />).toJSON()
  expect(tree).toMatchSnapshot()
})
