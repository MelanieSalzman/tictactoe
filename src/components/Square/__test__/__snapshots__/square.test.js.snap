// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square matches snapshot 1`] = `
<button
  className="square"
  data-testid="square"
/>
`;
